src/lsm/manifest_level.zig

```diff
@@ -815,28 +827,30 @@ pub fn TestContext(
         inserts: u64 = 0,
         removes: u64 = 0,
 
-        fn init(random: std.rand.Random) !Self {
-            var pool = try TestPool.init(
+        fn init(random: std.rand.Random) !ManifestLevel {
```

Comment:
Could we make this in-place as well?

I don't feel good about this-pattern, where we delegate from by-value init to in-place init. 

This is because with in-place, I think it would also be reasonable to assume that you aren't going to get moved, and to assume pointer stability. But if your caller moves you, there's no pointer stability! 

So the risk here is that at some point in the future, we'll notice that, for example, `ManifestLevelType` uses in-place init, use that to observe the pointer, and than the fuzz test would _continue_ to work because Zig optimizes the code to remove copies, so we don't notice the mistake right way. And then, during a Zig upgrade, its optimization change, it introduces the copy, and we'd have to chase that for a long time.

Should we maybe mention this in TigerStyle?

1. To construct large size-of objects, pass an out pointer
2. out-pointer inits can assume pointer stability and immovable types
3. but make sure that in-place ints are viral --- if your field is in-place initialized, you should be in-place as well!